client/utils/date-time.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { dateI18n } from '@wordpress/date';
+
+type DateTimeFormat = string | null;
+
+interface FormatDateTimeOptions {
+	includeTime?: boolean; // Whether to include time in the formatted string (defaults to true)
+	useGmt?: boolean; // Whether to display the time in GMT/UTC (defaults to false)
+	separator?: string; // Separator between date and time (defaults to ' / ')
+	customFormat?: DateTimeFormat; // Custom format to use instead of WordPress settings
+}
+
+/**
+ * Formats a date and time string according to WordPress settings or a custom format.
+ *
+ * @param { string | Date } dateTime - The date and time string (e.g., '2024-10-23 15:28:26') or a JS Date object.
+ * @param { FormatDateTimeOptions } options - Additional options to control time inclusion and whether to use GMT/UTC.
+ * @return { string } - The formatted date and time string.
+ */
+export function formatUserDateTime(
+	dateTime: string | Date,
+	options: FormatDateTimeOptions = {
+		includeTime: true,
+		useGmt: false,
+		separator: ' / ',
+		customFormat: null,
+	}
+): string {
+	const {
+		customFormat = null,
+		includeTime = true,
+		useGmt = false,
+		separator = ' / ',
+	} = options;
+
+	// Use the WordPress settings for date and time format if no custom format is provided
+	const format =
+		customFormat ||
```

Comment:
Should we validate the `customFormat` to ensure it is a valid option for the `dateI18n` function?

Reply:
I looked into adding this validation, but it turns out to be trickier than it seems. The issue is that the `dateI18n` function supports the same formats as [PHP’s date function](https://www.php.net/manual/en/function.date.php). According to the [documentation](https://www.php.net/manual/en/datetime.format.php), "Unrecognized characters in the format string will be printed as-is." This provides a lot of flexibility. For instance, unsupported characters can be included as long as they’re escaped: a format like 'l \t\h\e jS' can output something like "Wednesday the 19th."

Given the flexibility that dateI18n offers with custom formats, I don't think it makes sense to implement strict validation. Instead, I’ve added unit tests to document this behavior: [368e80e](https://github.com/Automattic/woocommerce-payments/pull/9635/commits/368e80e72c06e2777cb87bb5b9c40d60b62de473) and [666473d](https://github.com/Automattic/woocommerce-payments/pull/9635/commits/666473d233e287747e8fc427a14b6ea327ec36ad). What do you think?

Reply:
Fair and makes sense.